### PR TITLE
[HAML-Lint] Improve issue ID and links for RuboCop

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

HAML-Lint uses internally RuboCop. But HAML-Lint issue's ID is always `RuboCop`.
This change converts it to `RuboCop:{cop_name}` such as `RuboCop:RuboCop:Lint/UselessAssignment`
and adds a link to the RuboCop cop documentation.

I think this change will allow users to handle HAML-Lint issues more easily.

Also, this change extracts common methods for RuboCop to a new module: `Runners::RuboCopUtils`.

> Link related issues or pull requests.

This refactoring should be helpful for Slim-Lint (#1720).

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: test/smokes/haml_lint/expectations.rb [hunk 1]
```diff
@@ -78,8 +81,7 @@
   type: "success",
   issues: [
     {
-      message:
-        "Lint/UselessAssignment: Useless assignment to variable - `unused_variable`. (https://rubystyle.guide#underscore-unused-vars)",
+      message: "Useless assignment to variable - `unused_variable`. (https://rubystyle.guide#underscore-unused-vars)",
       links: %w[https://github.com/sds/haml-lint/blob/v0.34.0/lib/haml_lint/linter#rubocop],
       id: "RuboCop",
```
Comment (ybiquitous) [inline test/smokes/haml_lint/expectations.rb:86]: [note] When `DisplayCopNames: false` in `.rubocop.yml`, an issue ID is still `RuboCop`.

**Comment by ybiquitous** [comment]
The CI failure should be fixed via #2011.

**Review by yykamei — APPROVED** [review]
I see. This change is great :+1: 

I think it's a breaking change because some IDs will change. Especially regarding Sider, some closed issues of HAML-Lint might be reopened while applying this new version of Runners to Sider.

**Comment by ybiquitous** [comment]
Surely. Users may be temporarily inconvenienced, but there are not so many `RuboCop` issues recently when I look into them.

I think this is tolerable, right?

**Comment by yykamei** [comment]
Yes, I agree. It's a breaking change, but virtually, there seem to be few problems. I believe.

**Pull request merged**